backend/Tim.Backend/Controllers/External/QueryExternalController.cs

```diff
@@ -88,37 +55,27 @@ public class QueryExternalController : Controller
         [ProducesResponseType(StatusCodes.Status400BadRequest)]
         [ProducesResponseType(StatusCodes.Status204NoContent)]
         [ProducesResponseType(StatusCodes.Status500InternalServerError)]
-        public async Task<ActionResult<KustoQueryTriggerResponse>> ExecuteQuery(
-           [FromBody] KustoQueryExecuteRequest queryRequest,
+        public async Task<ActionResult<KustoQueryRun>> ExecuteQuery(
+           [FromBody] KustoQuery queryRequest,
            CancellationToken cancellationToken)
         {
             queryRequest.Validate();
 
+            var accessToken = await HttpContext.GetTokenAsync("Bearer", "access_token");
+
             var activityName = "QueryExternalController-executeQuery";
             try
             {
-                if (string.IsNullOrWhiteSpace(queryRequest.Query))
-                {
-                    return BadRequest(ProblemDetailsFactory.CreateProblemDetails(
-                        HttpContext,
-                        statusCode: StatusCodes.Status400BadRequest,
-                        title: "Kusto Query must be specified. "));
-                }
+                var kustoClient = await GetOBOKustoClient(queryRequest, accessToken);
+                var delayQueryRunner = new DelayedQueryRunner(kustoClient, m_dbRepository);
+                var queryRun = await delayQueryRunner.DelayRun(queryRequest, cancellationToken);
 
-                var queryRunRecord = await QueryRunHelper.ExecuteQueryHelper(queryRequest, m_dbClient, m_ingestClient, m_configs.Value.QueryRunsContainerName, cancellationToken);
-
-                // TODO 2022-10-07 - switch to app id/key for kusto queries - will revert back to user executes queries once we have a chance to refactor auth
-#pragma warning disable CS4014 // Because this call is not awaited, execution of the current method continues before the call is completed
-#pragma warning disable IDE0058 // Expression value is never used
-                Task.Run(() => m_kustoQueryExecutor.RunQuery(string.Empty, queryRunRecord, m_userReader, m_dbClient, m_kustoQueryClient, m_configs.Value, CancellationToken.None));
-#pragma warning restore IDE0058 // Expression value is never used
-#pragma warning restore CS4014 // Because this call is not awaited, execution of the current method continues before the call is completed
-                return new KustoQueryTriggerResponse(queryRunRecord.QueryRunId, "Query has been queued. ");
+                return queryRun;
             }
             catch (Exception ex)
             {
                 m_logger.Error(ex, "Failed to perform operation {operation} with exception: {exception}", activityName, ex);
-                return Problem(title: $"Query failed.", detail: ex.Message, statusCode: StatusCodes.Status400BadRequest);
+                return Problem(title: "Query failed.", detail: ex.Message, statusCode: StatusCodes.Status400BadRequest);
```

Comment:
it could mean that query failed or the request to trigger a query failed, do we want to be a bit more specific or the assumption is the exception will speak for itself?

Reply:
Good question. Most of the error handling happens within the QueryRunner and is passed into the QueryRun result. Whereas, this error occurs unexpectedly, so we can pass this onto the frontend.